Rename the tab model to Code Completion and add a description.

benedict-lee opened the pull request:
### change tab model name 
I think code completion is a more abstract and generic representation than the tab model.
Because tab is a way to accept code completion, it doesn't seem appropriate to call it by a formal name.

### update code completion model description
Added information about FIM(Fill in the Middle) and the newest model, codegemma, to the code completion description, as well as the parameter sizes for each model.

### Added llama3 as an example chat model.
It's not a code-specific LLM, but it has the highest human eval score of any open-source model.

TyDunn commented on where-we-are-today/code-completion.md:
```diff
@@ -0,0 +1,2 @@
+# Code Completion model
+The "code completion" model component is used to support autocomplete suggestions and is also referred to as the "tab-autocomplete model". Tab-autocomplete models use models trained with special templates, such as FiM(Fill in the Middle), that specialize in code infilling, and typically use small models, on the order of 1-15B for low latency. Because developers need a suggestion within 500ms, you ...
```
```suggestion
The "autocomplete" model component is used to power code completion suggestions and is typically a 1-15B parameter model. The models are run on your laptop or on a server and have generally been trained with special templates like fill-in-the-middle (FIM) for code infilling. Because developers need a suggestion within 500ms, you generally need to use a smaller model in order to meet the latency requirements. However, the quality of suggestions you get from models that are too small is bad. Thus, the tab-autocomplete model is optimized primarily with these two constraints in mind. Examples of models used for code completion include [Codex](https://arxiv.org/pdf/2107.03374.pdf), [CodeGemma](https://developers.googleblog.com/2024/04/gemma-family-expands.html), [Code Llama](https://arxiv.org/pdf/2308.12950.pdf), [DeepSeek Coder Base](https://deepseekcoder.github.io/), [StarCoder 2](https://arxiv.org/pdf/2402.19173.pdf), etc.
```

TyDunn reviewed (changes requested):
Thank you for this PR, @benedict-lee! 

I think that it is more beautiful for this component to be a single word, but I agree that "tab" is not the best one. I tried out "autocomplete" in my suggestions and then moved "code completion" to how we describe this component. We will need to rename the file from `code-completion.md` to `autocomplete.md` before we merge if we go this route. I also think we should link to research papers / blog posts for the models instead of enumerating all of the different model sizes. 

Let me know what you think. Looking forward to the next iteration!

benedict-lee commented:
@TyDunn Thanks for the review.
I agree with you, autocomplete is better than code-completion.
I've added all your suggestions, and renamed the file again.

TyDunn commented on where-we-are-today/autocomplete.md:
```diff
@@ -0,0 +1,2 @@
+# Autocomplete model
```
```suggestion
# Autocomplete model

```

TyDunn reviewed (approved):
Thank you for your contribution, @benedict-lee!
